sqlglot/executor/__init__.py

```diff
@@ -34,6 +37,20 @@ def execute(sql, schema, read=None):
     plan = Plan(expression)
     logger.debug("Logical Plan: %s", plan)
     now = time.time()
-    result = PythonExecutor().execute(plan)
+    result = PythonExecutor(tables=tables).execute(plan)
     logger.debug("Query finished: %f", time.time() - now)
     return result
+
+
+def _ensure_tables(tables):
+    result = {}
+    if not tables:
+        return result
+    for name, table in tables.items():
+        if isinstance(table, Table):
+            result[name] = table
+        else:
+            columns = tuple(table[0].keys()) if table else ()
```

Comment:
you don't need keys

Reply:
🤦 how many times have you told me this? lol